game/scripts/vscripts/Sections/Chapter5/SectionRoshan.ts

```diff
@@ -0,0 +1,270 @@
+import * as tut from "../../Tutorial/Core";
+import * as tg from "../../TutorialGraph/index";
+import { RequiredState } from "../../Tutorial/RequiredState";
+import { GoalTracker } from "../../Goals";
+import { chapter5Blockades, runeSpawnsLocations } from "./Shared";
+import { findRealPlayerID, getOrError, getPlayerHero, setUnitPacifist } from "../../util";
+import { modifier_custom_roshan_attack_speed } from "../../modifiers/modifier_custom_roshan_attack_speed";
+
+const sectionName: SectionName = SectionName.Chapter5_Roshan;
+
+let graph: tg.TutorialStep | undefined = undefined
+let canPlayerIssueOrders = false;
+
+const requiredState: RequiredState = {
+    requireSlacksGolem: true,
+    requireSunsfanGolem: true,
+    slacksLocation: Vector(-5906, -3892, 256),
+    sunsFanLocation: Vector(-5500, -4170, 256),
+    heroLocation: runeSpawnsLocations.topPowerUpRunePos.__add(Vector(-200, 0, -48)),
+    heroLocationTolerance: 800,
+    heroHasDoubleDamage: true,
+    blockades: [
+        chapter5Blockades.direJungleLowgroundRiver,
+        chapter5Blockades.topLaneRiver,
+        chapter5Blockades.radiantSecretShopRiver,
+        chapter5Blockades.direOutpostRiver,
+        chapter5Blockades.radiantAncientsRiver,
+        chapter5Blockades.radiantMidTopRiver,
+        chapter5Blockades.direMidTopRiver,
+        chapter5Blockades.midRiverTopSide,
+    ],
+};
+
+// Move to state when implementing final section of CH5
+const friendlyHeroesInfo = [
+    { name: "npc_dota_hero_tidehunter", loc: Vector(-2192, 1718, 0) }, // Offlane, tank
+    { name: "npc_dota_hero_juggernaut", loc: Vector(-2026, 1550, 0) }, // Carry
+    { name: "npc_dota_hero_mirana", loc: Vector(-2074, 1465, 0) }, // Position 4
+    { name: "npc_dota_hero_lion", loc: Vector(-2282, 1462, 0) }, // Position 5
+];
+
+const roshanMusic = "valve_ti10.music.roshan"
+
+function onStart(complete: () => void) {
+    print("Starting", sectionName);
+    const goalTracker = new GoalTracker();
+    const goalEnterRoshPit = goalTracker.addBoolean("Walk into Roshan's pit.");
+    const goalUpgradeTalents = goalTracker.addBoolean("Choose and pick between the available talents.")
+    const goalDefeatRoshan = goalTracker.addBoolean("Defeat Roshan!");
+    const goalPickupAegis = goalTracker.addBoolean("Pick up the Aegis of the Immortal.")
+    const goalLeaveRoshPit = goalTracker.addBoolean("Leave Roshan's pit and move to the next marker.")
+
+    const roshPitGoalPosition = Vector(-2600, 2200, 28)
+    const leaveRoshPitGoalPosition = Vector(-2140, 1740, 0)
+
+    const playerHero = getOrError(getPlayerHero())
+
+    const itemDaedalus = "item_greater_crit"
+    const itemAC = "item_assault"
+    const itemPowerTreads = "item_power_treads"
+    const itemHeart = "item_heart"
+    const itemAegis = "item_aegis"
+
+    let roshan = Entities.FindAllByName("npc_dota_roshan")[0] as CDOTA_BaseNPC
```

Comment:
Can be `const`

Reply:
I reassign it later in case the FindAllByName[0] returns nothing, so I have to use `let`

Reply:
Ah missed that